### PR TITLE
Refactor the way certs are uploaded and stub out rotate command

### PR DESCRIPTION
**Pull request by fieldju**
With all the highlander changes we don't store as much stuff in s3 any more and we need to be able to store multiple cert and key files to support rotation.

- Refactored config store to use one encrypted file for all data, since we no longer need to read things unencrypted, and remove unneeded methods

the data now looks like the following

	{
	  "database_password" : "Sup3rS3cr3tP@ssw0rd-On-The-Publ1c-Int3rn3t",
	  "metrics_topic_arn" : null,
	  "certificate_data" : [ {
	    "identity_management_certificate_name" : "ASCAXXXXXXXXXXXXXXXXX",
	    "not_before" : 1510892509000,
	    "not_after" : 1518668509000,
	    "uploaded" : 1510892509000,
	    "common_name" : "highlander-dev.example.com",
	    "subject_alternate_names" : [ "dev.example.com", "dev.us-east-1.example.com", "dev.us-west-2.example.com", "highlander-dev.example.com", "highlander-dev.us-west-2.example.com", "origin.dev.example.com", "origin.highlander-dev.example.com" ]
	  } ]
	}

- Make upload certs support multiple certs with keys. (folders of cert files with name as key)
  - cert and key files go into `/certificates/{identity_management_certificate_name}/`
- Delete misc classes that are not needed anymore
- Store more information about certs in env config rather than just the name, like what domains there good for and when they were uploaded and expire.
- Stub out the rotate composite command and upload cert to alb command for next iteration of work

### CODE REVIEW

File: src/main/java/com/nike/cerberus/cli/EnvironmentConfigToArgsMapper.java [hunk 1]
```diff
@@ -185,30 +184,10 @@ private EnvironmentConfigToArgsMapper() {
     }
 
     private static List<String> getUploadCertFilesCommandArgs(EnvironmentConfig environmentConfig, String[] passedArgs) {
```
Comment (sdford) [inline src/main/java/com/nike/cerberus/cli/EnvironmentConfigToArgsMapper.java]: It looks like passedArgs is not used here, should it be?

I see we were checking for --overwrite flag in the passedArgs. Are we still safeguarding against overwrite?

Comment (fieldju) [reply src/main/java/com/nike/cerberus/cli/EnvironmentConfigToArgsMapper.java]: You don't override certs anymore you can upload as many as you want and the latest will be used

Comment (fieldju) [reply src/main/java/com/nike/cerberus/cli/EnvironmentConfigToArgsMapper.java]: --overwrite has been removed from the command, so I can remove passed args here from the signature

File: src/main/java/com/nike/cerberus/operation/cms/CreateCmsClusterOperation.java [hunk 2]
```diff
@@ -75,11 +73,10 @@ public CreateCmsClusterOperation(final EnvironmentMetadata environmentMetadata,
     public void run(final CreateCmsClusterCommand command) {
         final String environmentName = environmentMetadata.getName();
         final VpcOutputs vpcOutputs = configStore.getVpcStackOutputs();
-        final Optional<String> cmsServerCertificateArn = configStore.getServerCertificateArn(Stack.CMS);
-        final Optional<String> pubKey = configStore.getCertPart(Stack.CMS, ConfigConstants.CERT_PART_PUBKEY);
+        final List<CertificateInformation> certInfoListForStack = configStore.getCertificationInformationList();
 
-        if (!cmsServerCertificateArn.isPresent() || !pubKey.isPresent()) {
-            throw new IllegalStateException("CMS certificate has not been uploaded!");
+        if (!certInfoListForStack.isEmpty()) {
```
Comment (sdford) [inline src/main/java/com/nike/cerberus/operation/cms/CreateCmsClusterOperation.java]: If it is empty we want to throw right?

Comment (fieldju) [reply src/main/java/com/nike/cerberus/operation/cms/CreateCmsClusterOperation.java]: Yes, I will edit that real quick

File: src/main/java/com/nike/cerberus/domain/environment/CertificateInformation.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    private DateTime notAfter;
+    private DateTime uploaded;
+    private String commonName;
+    private List<String> subjectAlternateNames;
+
+    public String getIdentityManagementCertificateName() {
+        return identityManagementCertificateName;
+    }
+
+    public void setIdentityManagementCertificateName(String identityManagementCertificateName) {
```
Comment (sdford) [inline src/main/java/com/nike/cerberus/domain/environment/CertificateInformation.java]: Is this just the name? Can we store the ARN here as well or instead?

Comment (fieldju) [reply src/main/java/com/nike/cerberus/domain/environment/CertificateInformation.java]: It just the name, as that is all that is used

File: src/main/java/com/nike/cerberus/command/core/AddCertificateToAlbCommand.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+import static com.nike.cerberus.command.core.AddCertificateToAlbCommand.COMMAND_NAME;
+
+@Parameters(
+        commandNames = {
+                COMMAND_NAME
+        },
+        commandDescription = "Adds a new cert to the alb"
+)
+public class AddCertificateToAlbCommand implements Command {
```
Comment (sdford) [inline src/main/java/com/nike/cerberus/command/core/AddCertificateToAlbCommand.java]: Can the ALB contain more than one certificate?

If not, then I'm not sure this command is needed. You should be able to update the ALB stack with the new IAM certificate ARN for CMS.

Comment (fieldju) [reply src/main/java/com/nike/cerberus/command/core/AddCertificateToAlbCommand.java]: The ALB does support more than one cert or rotating would be impossible without downtime, but this is just a stub, might change it around.

Comment (sdford) [reply src/main/java/com/nike/cerberus/command/core/AddCertificateToAlbCommand.java]: Cool!

File: src/main/java/com/nike/cerberus/operation/core/CreateLoadBalancerOperation.java [hunk 5]
```diff
@@ -65,8 +63,9 @@ public void run(final CreateLoadBalancerCommand command) {
         final String environmentName = environmentMetadata.getName();
         final VpcOutputs vpcOutputs = configStore.getVpcStackOutputs();
 
-        final String sslCertificateArn = configStore.getServerCertificateArn(Stack.CMS)
-                .orElseThrow(() -> new IllegalStateException("Could not retrieve SSL certificate ARN!"));
+        // Use latest cert, if there happens to be more than one for some reason
+        final String sslCertificateArn = configStore.getCertificationInformationList()
```
Comment (sdford) [inline src/main/java/com/nike/cerberus/operation/core/CreateLoadBalancerOperation.java]: Does this work with the ALB CloudFormation? From what I understand in your previous comment this method will return the certificate name, not the ARN.

At the least we should rename this to the cert name

File: src/main/java/com/nike/cerberus/domain/environment/CertificateInformation.java (hunk 3 above)
Comment (fieldju) [reply src/main/java/com/nike/cerberus/domain/environment/CertificateInformation.java]: Your right, I missed that there was a conversion in the old code, fixed!

File: src/main/java/com/nike/cerberus/operation/core/CreateLoadBalancerOperation.java (hunk 5 above)
Comment (fieldju) [reply src/main/java/com/nike/cerberus/operation/core/CreateLoadBalancerOperation.java]: fixed

**Pull request merged**